lib/procinfo_mac.cpp

```diff
@@ -167,3 +169,42 @@ int procinfo_setup(PROC_MAP& pm) {
     setlocale(LC_ALL, old_locale.c_str());
     return 0;
 }
+
+// get total user-mode CPU time
+//
+// From usr/include/mach/processor_info.h:
+// struct processor_cpu_load_info {             /* number of ticks while running... */
+//	 unsigned int    cpu_ticks[CPU_STATE_MAX]; /* ... in the given mode */
+// };
+//
+double total_cpu_time() {
+    static natural_t processorCount = 0;
+    processor_cpu_load_info_t cpuLoad;
+    mach_msg_type_number_t processorMsgCount;
+    static double scale;
```

Comment:
There is no default value for `scale`.
In case `processorCount` is not equal zero, this value will be set to `0` be default, so later this function will always return `0` because the last statement is multiplication with `scale`.
If this is intended - then fine, but I believe this is a mistake

Reply:
P.S. If you want just to initialize this static variable once, then it's better to have either some flag or just verify if this value is zero (and then initialize it with a proper value) otherwise just continue execution

Reply:
@AenBleidd I intended this. I have no better solution in case of such errors because the code by @davidpanderson ( line 495 in _client/app.cpp_) which calls this function has no ability to handle errors returned from `total_cpu_time()`. The UNIX / Linux version of `total_cpu_time()` created for #4859 in _lib/procinfo_unix.cpp_ also returns 0 in case of any error opening or parsing `/proc/stat`. The MSW version of `total_cpu_time()` in _lib/procinfo_win.cpp_ from #4859 does no error checking at all.

I agree that the code which calls this should take some sort of appropriate action if `total_cpu_time()` indicates an error by returning 0.

Reply:
Ok, code looks OK to me. Please let me know when it's tested and ready to merge

Reply:
One possibility for handling errors from `total_cpu_time()` might be to revert to the old method of calculating non_boinc CPU usage (lines 502 - 523 in _client/app.cpp_).

Reply:
> P.S. If you want just to initialize this static variable once, then it's better to have either some flag or just verify if this value is zero (and then initialize it with a proper value) otherwise just continue execution

I'm not sure I understand what you are suggesting. `processorCount ` is always updated by the `host_processor_info()` call; there is no way to avoid that. But I am having it do double duty, both as a flag allowing me to call `sysconf(_SC_CLK_TCK)` only once, and also to indicate an error by returning 0 if for some reason `processorCount ` is 0 (as you pointed out in the first comment above.)

Reply:
This double duty is exactly that I don't like: it's an additional complication and fragile point, and in case you need to change the code, you still should remember that this variable serves as a flag too.
I believe in this case it's acceptable but in general case I'd recommend to use a separate variable as a flag.